File: _includes/sort-benchmark/algorithms/postman.rs
fn postman_sort_bytes(a: &mut [usize], byte: usize) {
    const W: usize = 256;
    const THRESHOLD: usize = 16;

    if a.len() <= THRESHOLD {
        insertion_sort(a);
        return;
    }
    if byte >= std::mem::size_of::<usize>() {
        return;
    }

    let shift = (std::mem::size_of::<usize>() - 1 - byte) * 8;
    let mut buckets: [Vec<usize>; W] = std::array::from_fn(|_| Vec::new());

    for &value in a.iter() {
        let digit = ((value >> shift) & 0xFF) as usize;
        buckets[digit].push(value);
    }

    let mut offset = 0usize;
    for bucket in buckets.iter_mut() {
        if bucket.len() > 1 {
            postman_sort_bytes(bucket, byte + 1);
        }
        let len = bucket.len();
        if len > 0 {
            a[offset..offset + len].copy_from_slice(bucket);
            offset += len;
        }
    }
}

fn postman_sort(a: &mut [usize]) {
    if !a.is_empty() {
        postman_sort_bytes(a, 0);
    }
}

File: _includes/sort-benchmark/algorithms/postman.swift
fileprivate func postman_sort_bytes(_ a: UnsafeMutableBufferPointer<Int>, _ byte: Int) {
    let W = 256
    let THRESHOLD = 16

    if a.count <= THRESHOLD {
        insertion_sort(a)
        return
    }
    if byte >= MemoryLayout<Int>.size {
        return
    }

    let shift = (MemoryLayout<Int>.size - 1 - byte) * 8
    var buckets = [[Int]](repeating: [], count: W)

    for i in 0..<a.count {
        let value = a[i]
        let digit = (value >> shift) & 0xFF
        buckets[digit].append(value)
    }

    var offset = 0
    for b in 0..<W {
        if buckets[b].count > 1 {
            buckets[b].withUnsafeMutableBufferPointer { postman_sort_bytes($0, byte + 1) }
        }
        let len = buckets[b].count
        if len > 0 {
            for i in 0..<len {
                a[offset + i] = buckets[b][i]
            }
            offset += len
        }
    }
}

func postman_sort(_ a: inout [Int]) {
    a.withUnsafeMutableBufferPointer { postman_sort($0) }
}

func postman_sort(_ a: UnsafeMutableBufferPointer<Int>) {
    if !a.isEmpty {
        postman_sort_bytes(a, 0)
    }
}
